tools/interface/ontology.py

```diff
@@ -0,0 +1,61 @@
+import sys
+from typing import List, NamedTuple
+
+sys.path.insert(1, '../validators/ontology_validator/')
+
+from yamlformat.validator.entity_type_lib import EntityType
+from yamlformat.validator.presubmit_validate_types_lib import ConfigUniverse
+from yamlformat import constants
+
+StandardField = NamedTuple('StandardField', [('namespace', str), ('name', str), ('is_required', bool)])
```

Comment:
"StandardField" is a misnomer for this thing. A "standard field" can't have optionality because it's defined outside of the context of an entitytype.  It's also ambiguous whether or not an increment can be part of the name.  

I think the approach that might be easier to use and more flexible in the future is to create an `EntityTypeField` class that has properties and accessor methods for all of the relevant concepts:
-namespace
-name
-increment
-optionality

In the API you can follow the convention that a specified property is a strict "must" have and an unspecified property (==None) is a wildcard.   For example:

`EntityType(namespace = 'HVAC', name='run_status')`  would match`run_status` as an optional or required field.  It would also match `run_status_1`.  

`EntityType(namespace = 'HVAC', name='run_status', increment='', is_optional=False)` would only match required fields having `run_status` as the name.

You can similarly have a `StandardField` object with only `namespace`, `name` if you need it.  You can then use inheritance to do nice stuff like any `EntityTypeField` is also a `StandardField`.  For example, the `EntityTypeField` representing HVAC/run_status_1, optional is also the Standard Field HVAC/run_status.

Reply:
I like the idea of having an `EntityTypeField` class extended from some base field class. What stopped me from doing this before was that the `Field` class already exists and creating another similar class would be redundant and could lead to performance issues.